crates/turbo-tasks-memory/src/task_stats.rs

```diff
@@ -0,0 +1,123 @@
+use std::{
+    sync::atomic::{AtomicBool, Ordering},
+    time::Duration,
+};
+
+static ENABLE_FULL_STATS: AtomicBool = AtomicBool::new(false);
+
+/// Enables full stats collection.
+///
+/// This is useful for debugging, but it has a slight memory and performance
+/// impact.
+pub fn enable_full_stats() {
+    ENABLE_FULL_STATS.store(true, Ordering::Release);
+}
+
+/// Returns `true` if full stats collection is enabled.
+pub fn full_stats() -> bool {
+    ENABLE_FULL_STATS.load(Ordering::Acquire)
+}
```

Comment:
Could we avoid the global and use a method on `TurboTasks` instead?

Reply:
This is possible, but it will incur one dynamic dispatch per task-stats creation.

Reply:
Also, it's somewhat awkward to get access to the turbo tasks instance where we need it (`TaskStats` instantiation). We'd have to pass it down two function calls.

Reply:
Fixed in https://github.com/vercel/turbo/pull/2765/commits/01017540e85d757bfc8b69086fb04671a0b59b2b